### PR TITLE
GH-2443: Distinct Concurrency for Retry Containers

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves https://github.com/spring-projects/spring-kafka/issues/2443

Optionally configure a different concurrency for retry containers.

### CODE REVIEW

**Comment by garyrussell** [comment]
cc/ @tomazfernandes

File: spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java [hunk 1]
```diff
@@ -93,6 +95,11 @@ public RetryTopicConfigurationBuilder dltHandlerMethod(String beanName, String m
 		return this;
 	}
 
+	public RetryTopicConfigurationBuilder concurrency(Integer concurrency) {
```
Comment (tomazfernandes) [inline spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java:98]: I know most of the methods in this class don't have proper javadocs, but perhaps we should at least include in the new ones?

**Review by tomazfernandes — COMMENTED** [review]
LGTM, just one minor comment

File: spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java (hunk 1 above)
Comment (tomazfernandes) [reply spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java:98]: BTW, do you think it'd be worth it adding javadocs for the other methods as well? I can open a PR with that if you'd like.

File: spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java [hunk 2]
```diff
@@ -190,15 +254,37 @@ public RetryTopicConfigurationBuilder maxAttempts(int maxAttempts) {
 		return this;
 	}
 
+	/**
```
Comment (tomazfernandes) [inline spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java:257]: Maybe we should add that this is in milliseconds? Had to go through the code to figure out 😄

File: spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java [hunk 3]
```diff
@@ -138,50 +157,95 @@ public RetryTopicConfigurationBuilder autoStartDltHandler(@Nullable Boolean auto
...
+	 */
 	public RetryTopicConfigurationBuilder setTopicSuffixingStrategy(TopicSuffixingStrategy topicSuffixingStrategy) {
 		this.topicSuffixingStrategy = topicSuffixingStrategy;
 		return this;
 	}
 
 	/* ---------------- Configure BackOff -------------- */
 
+	/**
+	 * Configure the maximum delivery attempts.
```
Comment (tomazfernandes) [inline spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java:245]: Maybe add that this includes the first attempt? Or perhaps it's not necessary?

**Review by tomazfernandes — COMMENTED** [review]
Awesome, thanks!

Just a couple comments

**Review by tomazfernandes — APPROVED** [review]
Looks great, thanks!

**Pull request merged**